src/core/src/op/gru_sequence.cpp

```diff
@@ -80,91 +60,23 @@ void op::v5::GRUSequence::validate_and_infer_types() {
-    }
-
-    NODE_VALIDATION_CHECK(this,
-                          Dimension::merge(merged_num_directions, merged_num_directions, valid_num_directions),
-                          "Parameter 'num_directions' doesn't match with direction '",
-                          m_direction,
-                          "' in GRUSequence. Expected ",
-                          valid_num_directions,
-                          ", actual ",
-                          merged_num_directions);
-
-    // Validate hidden_size value for W, R, B inputs
-    if (merged_hidden_size.is_static()) {
-        if (w_pshape[1].is_static()) {
-            NODE_VALIDATION_CHECK(this,
-                                  w_pshape[1].compatible(merged_hidden_size * gru_seq_gates_count),
-                                  "Parameter hidden_size mistmatched in W input. Current value is: ",
-                                  w_pshape[1].get_length(),
-                                  ", expected: ",
-                                  merged_hidden_size.get_length() * gru_seq_gates_count,
-                                  ".");
-        }
-
-        if (r_pshape[1].is_static()) {
-            NODE_VALIDATION_CHECK(this,
-                                  r_pshape[1].compatible(merged_hidden_size * gru_seq_gates_count),
-                                  "Parameter hidden_size mistmatched in R input. Current value is: ",
-                                  r_pshape[1].get_length(),
-                                  ", expected: ",
-                                  merged_hidden_size.get_length() * gru_seq_gates_count,
-                                  ".");
-        }
-
-        if (b_pshape[1].is_static()) {
-            NODE_VALIDATION_CHECK(
-                this,
-                b_pshape[1].compatible(merged_hidden_size *
-                                       (m_linear_before_reset ? (gru_seq_gates_count + 1) : gru_seq_gates_count)),
-                "Parameter hidden_size mistmatched in B input. Current value is: ",
-                b_pshape[1].get_length(),
-                ", expected: ",
-                merged_hidden_size.get_length() *
-                    (m_linear_before_reset ? (gru_seq_gates_count + 1) : gru_seq_gates_count),
-                ".");
-        }
-    }
+    const auto& x_pshape = get_input_partial_shape(0);
+    const auto& ht_pshape = get_input_partial_shape(1);
+    const auto& sl_pshape = get_input_partial_shape(2);
+    const auto& w_pshape = get_input_partial_shape(3);
+    const auto& r_pshape = get_input_partial_shape(4);
+    const auto& b_pshape = get_input_partial_shape(5);
 
+    std::vector<ov::PartialShape> input_shapes = {x_pshape, ht_pshape, sl_pshape, w_pshape, r_pshape, b_pshape};
+    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape{4}, ov::PartialShape{3}};
+    shape_infer(this, input_shapes, output_shapes);
     // Mark inputs which are relevant to output parameters
     for (size_t i = 0; i <= 5; ++i)
```

Comment:
the same here

Reply:
Ok, done. I've removed `set_input_is_relevant_to_shape(i);`